python-spec/src/somacore/spatial.py

```diff
@@ -0,0 +1,774 @@
+                for the particular SOMA implementation for details.
+
+        Returns:
+            A :class:`SpatialRead` with :class:`ReadIter` of :class:`pa.Table`s data.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def write(
+        self,
+        values: Union[pa.RecordBatch, pa.Table],
+        *,
+        platform_config: Optional[options.PlatformConfig] = None,
+    ) -> Self:
+        """Writes the data from an Arrow table to the persistent object.
+
+        As duplicate index values are not allowed, index values already present
+        in the object are overwritten and new index values are added.
+
+        Args:
+            values: An Arrow table containing all columns, including
+                the index columns. The schema for the values must match
+                the schema for the ``DataFrame``.
+
+        Returns: ``self``, to enable method chaining.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    # Metadata operations
+
+    @property
+    @abc.abstractmethod
+    def schema(self) -> pa.Schema:
+        """The schema of the data in this dataframe.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def index_column_names(self) -> Tuple[str, ...]:
+        """The names of the index (dimension) columns.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def axis_names(self) -> Tuple[str, ...]:
+        """The names of the axes of the coordinate space the data is defined on.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def coordinate_space(self) -> Optional[coordinates.CoordinateSpace]:
+        """Coordinate space for this geometry dataframe.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @coordinate_space.setter
+    @abc.abstractmethod
+    def coordinate_space(self, value: coordinates.CoordinateSpace) -> None:
+        """Coordinate space for this geometry dataframe.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def domain(self) -> Tuple[Tuple[Any, Any], ...]:
+        """The allowable range of values in each index column.
+
+        Returns: a tuple of minimum and maximum values, inclusive,
+            storable on each index column of the dataframe.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+
+class MultiscaleImage(  # type: ignore[misc]  # __eq__ false positive
+    base.SOMAObject,
+    Generic[_DenseND, _RootSO],
+    MutableMapping[str, _DenseND],
+    metaclass=abc.ABCMeta,
+):
+    """A multiscale image with an extendable number of levels.
```

Comment:
```suggestion
    """A multiscale image with an extendable number of resolution levels.
```

Trying to make it clearer to users what a "level" is in this context.